core/trino-main/src/main/java/io/trino/operator/TopNOperator.java

```diff
@@ -190,12 +204,18 @@ public void addInput(Page page)
     @Override
     public void finish()
     {
+        isFlushing = false;
         pageBuffer.finish();
     }
 
     private void addPage(Page page)
     {
+        checkState(!topNProcessor.isBuilderFull(), "TopN buffer is already full");
         topNProcessor.addInput(page);
+        if (partial && topNProcessor.isBuilderFull()) {
```

Comment:
Isn't "topNProcessor.isBuilderFull()" guaranteed to be false due to the checkState above?

Reply:
It can be full after "addInput".